test/JsonApiDotNetCoreMongoDbExampleTests/IntegrationTests/ReadWrite/Updating/Resources/UpdateResourceTests.cs

```diff
@@ -0,0 +1,502 @@
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using FluentAssertions;
+using JsonApiDotNetCore.Configuration;
+using JsonApiDotNetCore.MongoDb;
+using JsonApiDotNetCore.MongoDb.Repositories;
+using JsonApiDotNetCore.Resources;
+using JsonApiDotNetCore.Serialization.Objects;
+using Microsoft.Extensions.DependencyInjection;
+using MongoDB.Driver;
+using Xunit;
+
+namespace JsonApiDotNetCoreMongoDbExampleTests.IntegrationTests.ReadWrite.Updating.Resources
+{
+    public sealed class UpdateResourceTests
+        : IClassFixture<IntegrationTestContext<TestableStartup>>
+    {
+        private readonly IntegrationTestContext<TestableStartup> _testContext;
+        private readonly WriteFakers _fakers = new WriteFakers();
+
+        public UpdateResourceTests(IntegrationTestContext<TestableStartup> testContext)
+        {
+            _testContext = testContext;
+            
+            _testContext.RegisterResources(builder =>
+            {
+                builder.Add<RgbColor, string>();
+                builder.Add<UserAccount, string>();
+                builder.Add<WorkItem, string>();
+            });
```

Comment:
You can get rid of these everywhere by using auto-discovery. To do that:
1. In JsonApiDotNetCoreMongoDbExample.Startup, replace:
```c#
resources: builder => { .. }
```
with:
```c#
discovery: facade => facade.AddCurrentAssembly()
```
2. In IntegrationTestContext.CreateFactory, replace ```resources: _registerResources``` with the same. Next, you can remove the _registerResources callback and its method.
3. You _could_ do the same for the GettingStarted project, but I think its nice to keep that one as-is, so we demonstrate both possibilities.